### DISCUSSION TITLE
why internal image cache just work for "asset.rootDirectory.length > 0" condition？

### DISCUSSION THREAD

**Discussion by zjhao-Bytedance** [airbnb/lottie-ios#2299] [Q&A]
```
   if ([asset.imageName hasPrefix:@"data:"]) {         
       ...
    } else if (asset.rootDirectory.length > 0) { 
      ...
      id<LOTImageCache> imageCache = [LOTCacheProvider imageCache];
      if (imageCache) {
        image = [imageCache imageForKey:imagePath];
        if (!image) {
          image = [UIImage imageWithContentsOfFile:imagePath];
          [imageCache setImage:image forKey:imagePath];
        }
      } else {
        image = [UIImage imageWithContentsOfFile:imagePath];
      }
    } else {
      ...
    } 
```

  would like to understand why the built-in cache only supports the second condition (if (asset.rootDirectory.length > 0)) and if there's a specific reason for this choice. Thank you very much for answering this question.

**Comment by calda** [comment]
This looks like really old Objective-C code from Lottie 2.x. I'd suggest updating to a more recent version like Lottie 4.4.0 if possible!

**Reply by zjhao-Bytedance** [reply]
Thank you very much for your response. However, our project utilizes the Objective-C version of Lottie, and we are quite curious about why internal caching is only used in the second condition. Looking forward to your relevant answer.
